src/macaron/slsa_analyzer/checks/check_result.py

```diff
@@ -84,7 +147,8 @@ def get_summary(self) -> dict:
             "check_id": self.check.check_id,
             "check_description": self.check.check_description,
             "slsa_requirements": [str(BUILD_REQ_DESC.get(req)) for req in self.check.eval_reqs],
-            "justification": self.result.justification,
+            # The justification report is stored in a heapq where the first element has the highest confidence score.
+            "justification": self.result.justification_report[0][1],
```

Comment:
I don't think I'm convinced that a heap should be used here when we have to do things like overriding the comparator while the only thing we need the heap for currently is to get the justification with the highest score. It may be that you have other goals with this heap in mind for later features. Otherwise, I'm not sure how the heap is useful unless we need to maintain a priority queue while allowing both pushing and popping operations throughout its lifetime.

Wouldn't it be simpler just to either sort the list before returning from the property function, or use the `max` function like so?
```suggestion
            "justification": max(self.result.justification_report, key=lambda e: e[0])[1],
```
I also don't think performance-wise we gain anything with a heap.

Reply:
I agree that if the heap is only used to obtain the max value once after all insertions are done then there is no benefit to using a heap. Using `max` keyed on the confidence score would be sufficient, and allows removing the `__lt__` override (and I don't like that override since it is overly specific to a specific usage, as evidenced by the lengthy comment that is required to explain it by referencing how it is used, which is undesirable as definitions ideally should stand on their own). Though if you really needed the heap, you can still get rid of the `__lt__` override by using a wrapper (defined at the usage) that compares by inverse confidence score instead.

Reply:
See commit [d25bf57](https://github.com/oracle/macaron/pull/620/commits/d25bf57f3e6f57d29cf4c311207dbcfe1302727e).